app/web_framework/cluster_master_application.py

```diff
@@ -253,3 +255,15 @@ def get(self):
         self.write({
             'events': analytics.get_events(since_timestamp, since_id),
         })
+
+
+class _SlaveShutdownHandler(_ClusterMasterBaseHandler):
+    @authenticated
+    def post(self):
+        shutdown_all = self.decoded_body.get('shutdown_all')
+        slaves_to_shutdown = self._cluster_master.all_slaves_by_id().keys() if shutdown_all else\
+            [int(slave_id) for slave_id in self.decoded_body.get('slaves')]
+
+        for slave_id in slaves_to_shutdown:
+            slave = self._cluster_master.get_slave(slave_id)
```

Comment:
@josephharrington nope, I was saying that I thought the current behavior was to 500, but I didn't know that an ItemNotFoundError exception translated into a 404.